### PR TITLE
[JENKINS-33416] add support for all, allRelease and release jobs filters

### PR DESCRIPTION
**Pull request by ldez**
Related to https://issues.jenkins-ci.org/browse/JENKINS-33416

### CODE REVIEW

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/views/JobFiltersContext.groovy [hunk 1]
```diff
@@ -18,19 +18,29 @@ class JobFiltersContext extends AbstractContext {
     }
 
     /**
-     * Adds a job status filter.
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/views/JobFiltersContext.groovy]: Do not move methods. Keep the diff as small as possible.

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/views/JobFiltersContext.groovy [hunk 2]
```diff
@@ -18,19 +18,29 @@ class JobFiltersContext extends AbstractContext {
...
-            includeExcludeTypeString(statusesFilter.matchType.value)
-            Status.values().each { status ->
-                "${status.name().toLowerCase()}"(statusesFilter.status.contains(status))
-            }
-        }
+    /**
+     * Adds an all release job filter.
+     * Adds to the view all the jobs using the 'release' build wrapper.
+     */
+    @RequiresPlugin(id = 'release')
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/views/JobFiltersContext.groovy]: Add the `minimumVersion` attribute for new methods.

File: job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/views/ListViewJobFilterSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,73 @@
+package javaposse.jobdsl.dsl.views
+
+import javaposse.jobdsl.dsl.JobManagement
+import spock.lang.Specification
+
+class ListViewJobFilterSpec<T extends ListView> extends Specification {
```
Comment (daspilker) [inline job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/views/ListViewJobFilterSpec.groovy]: Move the tests to `ListViewSpec`.

File: job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/views/ListViewJobFilterSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        def filters = view.node.jobFilters[0].value()
+        filters.size() == 1
+
+        Node allJobsFilter = filters[0]
+        allJobsFilter.name() == 'hudson.views.AllJobsFilter'
+        allJobsFilter.children().size() == 0
+
+        1 * jobManagement.requirePlugin('view-job-filters')
+
+        where:
```
Comment (daspilker) [inline job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/views/ListViewJobFilterSpec.groovy]: do not use a data table when there is only one row

**Comment by ldez** [comment]
remarks considered, corrections done.

File: job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/views/ListViewSpec.groovy [hunk 5]
```diff
@@ -310,56 +310,56 @@ class ListViewSpec<T extends ListView> extends Specification {
 
         where:
         filter || children
-        { ->
-            status {
-                matchType(INCLUDE_UNMATCHED)
-                status(Status.UNSTABLE, Status.STABLE, Status.ABORTED, Status.FAILED, Status.DISABLED)
-            }
-        } | [
+                { ->
```
Comment (ldez) [inline job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/views/ListViewSpec.groovy]: sorry for this but I use the default IntelliJ like is defined in contributor's guide.
You want I revert indent modifications ?

Comment (daspilker) [inline job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/views/ListViewSpec.groovy]: Yes, please revert the modifications. Make sure to select the "Only VCS changed text" option in the reformat code dialog (crtl+alt+shift+L).

**Comment by ldez** [comment]
fix indent.

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/views/JobFiltersContext.groovy [hunk 6]
```diff
@@ -47,4 +47,31 @@ class JobFiltersContext extends AbstractContext {
             delegate.regex(regexFilter.regex)
         }
     }
+
+    /**
+     * Adds an all job filter.
+     */
+    @RequiresPlugin(id = 'view-job-filters')
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/views/JobFiltersContext.groovy]: add a minimum version

File: job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/views/ListViewSpec.groovy [hunk 7]
```diff
@@ -393,6 +393,57 @@ class ListViewSpec<T extends ListView> extends Specification {
         RegexMatchValue.NAME | '.*'
     }
 
+    def 'allJobsFilter'() {
+        when:
+        view.jobFilters { ->
```
Comment (daspilker) [inline job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/views/ListViewSpec.groovy]: remove the `->`

File: job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/views/ListViewSpec.groovy [hunk 8]
```diff
@@ -393,6 +393,57 @@ class ListViewSpec<T extends ListView> extends Specification {
         RegexMatchValue.NAME | '.*'
     }
 
+    def 'allJobsFilter'() {
```
Comment (daspilker) [inline job-dsl-core/src/test/groovy/javaposse/jobdsl/dsl/views/ListViewSpec.groovy]: You do not need to use camel case in Spock test method names. Rename this to `all jobs filter`

**Comment by ldez** [comment]
remarks considered, corrections done.

**Pull request merged**